### ISSUE TITLE
MultiLayerPerceptron gradient correct or not?

### ISSUE THREAD

**Issue by garibaldu** [garibaldu/comp421#1]
Curious about the behaviour of MultiLayerPerceptron - the cost looks like it drops but it's stochastic - often goes up. I thought this would be fixed if I lower the learning rate eta to really small, but it makes it worse! So I'm confused.

**Comment by Quirl** [comment]
May be a feature of the loss function? Escaping local maxima? Would explain why lowering eta makes it worse - it has to climb down the hill, but at a slower pace. My two cents.

**Comment by MashallAryan1** [comment]
BackProp code is correct. the problem was: sum of errors (not squared error) was shown in the plot. fixed it.
